habitat_sim/simulator.py

```diff
@@ -167,34 +185,16 @@ def _config_pathfinder(self, config: Configuration):
         self.pathfinder.seed(config.sim_cfg.random_seed)
 
     def reconfigure(self, config: Configuration):
-        assert len(config.agents) > 0
+        self._sanitize_config(config)
 
-        config.sim_cfg.create_renderer = any(
-            map(lambda cfg: len(cfg.sensor_specifications) > 0, config.agents)
-        )
-        config.sim_cfg.load_semantic_mesh = any(
-            map(
-                lambda cfg: any(
-                    map(
-                        lambda sens_spec: sens_spec.sensor_type == SensorType.SEMANTIC,
-                        cfg.sensor_specifications,
-                    )
-                ),
-                config.agents,
-            )
-        )
+        if self.config != config:
+            self.__set_from_config(config)
+            self.config = config
 
-        if self.config == config:
-            return
-
-        # NB: Configure backend last as this gives more time for python's GC
-        # to delete any previous instances of the simulator
-        # TODO: can't do the above, sorry -- the Agent constructor needs access
-        # to self._sim.get_active_scene_graph()
+    def __set_from_config(self, config: Configuration):
         self._config_backend(config)
         self._config_agents(config)
         self._config_pathfinder(config)
-        self.frustum_culling = config.sim_cfg.frustum_culling
```

Comment:
Does this get set anywhere now? Looks like the config setting is used directly in reconfigure to decide whether or not to split the mesh for the semantic SceneGraph in C++ [here](https://github.com/facebookresearch/habitat-sim/blob/e31c39fbf09655c434d32500395b5c1e46db7120/src/esp/sim/Simulator.cpp#L184). However, it seems that we don't consume the config setting for RGBD culling now.

Reply:
I looked over the code base and nothing was using that value, may have just been there from testing and didn't get properly removed before landing.

Reply:
I see. 👍  Maybe this is now tangential to the current change, but should this have been used [here](https://github.com/facebookresearch/habitat-sim/blob/d28b051eb72196365a2d03630f20b7d86599a268/habitat_sim/simulator.py#L405)? @bigbike

Reply:
I initially introduced in this PR:
https://github.com/facebookresearch/habitat-sim/commit/9a1fcb2c0b80d2d65a7b4f78cb476d719d71b34c

Yes, it is used in the place as you pointed out. @aclegg3 

I did it because I need to enable the frustum culling when the reconfigure() is called. It is not something just for testing and is left there accidentally.

If you removed it, where would it (enabling the FC or not) get reset?

Reply:
I miss-remember why I removed it.  I left it there accidentally in #620.  We have a property from the cpp bindings called `frustum_culling`, so `self.frustum_culling` just calls that property after #620.  I should have removed that call in that PR and moved it to cpp (makes sense to set that in the cpp `reconfigure` since its part of the backend/cpp config).

Reply:
Scratch that, I am just miss-remembering my line of thought entirely here.  I am just going to leave this the way it was.  Maybe doing in cpp would be better, but not worth changing.